daly-bms-uart.cpp

```diff
@@ -39,63 +40,277 @@ bool Daly_BMS_UART::Init()
     return true;
 }
 
-bool Daly_BMS_UART::getPackMeasurements(float &voltage, float &current, float &SOC)
+bool Daly_BMS_UART::update()
 {
-    this->sendCommand(COMMAND::VOUT_IOUT_SOC);
+
+    getPackMeasurements(); //0x90
+    getMinMaxCellVoltage(); //0x91
+    getPackTemp();  //0x92
+    getDischargeChargeMosStatus(); //0x93
+    getStatusInfo(); //0x94
+    //getCellVoltages(); //dont work, the answer string from BMS is too long and must splitet
+    getFailureCodes(); //0x98
+    /**
+ * put here the function call to recive all data one by one
+ * check if cell number ar set and then ask for cell data
+ * 
+ */
 
     if (!this->receiveBytes())
```

Comment:
`receiveBytes()` is already called in each of the individual command functions. I would expect this call to return false since we already pulled all the bytes from the input buffer in `getFailureCodes()`

Reply:
ok so remove?